backend/src/main/java/com/htecgroup/skynest/service/UserService.java

```diff
@@ -49,4 +49,7 @@ void demoteUser(
           UUID uuid);
 
   void disableUser(@Valid @UserNotAdmin UUID userId);
+
+  void addCompanyForUser(
+      @Valid @UserHasCompany UUID uuid, @Valid @AddUserInAdminCompany UUID companyId);
```

Comment:
so the user must not already be in a company, and admins can only add users to their own company

shouldn't it be something like UserNotInACompany instead of UserHasCompany?

and AdminIsInCompany or CurrentUserCompany instead of AddUserInAdminCompany?

Reply:
The first one I agree with, the second one you I think it's still missing something, maybe AdminAndUserInSameCompany

Reply:
the validator checks whether the current user, an admin, is in the company that's in the request

calling it AdminAndUserInSameCompany makes it seem like the user is already in that company, or at least that the annotation checks that

Reply:
based on what the validator checks, CurrentUserIsInCompany probably makes sense